lib/puppetserver/ca/puppet_config.rb

```diff
@@ -84,60 +84,68 @@ def resolve_settings(overrides = {})
         substitutions = Hash.new {|h, k| k }
         settings = {}
 
-        confdir = user_specific_conf_dir
-        settings[:confdir] = substitutions['$confdir'] = confdir
-
-        ssldir = overrides.fetch(:ssldir, '$confdir/ssl')
-        settings[:ssldir] = substitutions['$ssldir'] = ssldir.sub('$confdir', confdir)
-
-        certdir = overrides.fetch(:certdir, '$ssldir/certs')
-        settings[:certdir] = substitutions['$certdir'] = certdir.sub(unresolved_setting, substitutions)
-
-        cadir = overrides.fetch(:cadir, '$ssldir/ca')
-        settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
-
-        settings[:certname] = substitutions['$certname'] = overrides.fetch(:certname, @certname)
-
-        server = overrides.fetch(:server, '$certname')
-        settings[:server] = substitutions['$server'] = server.sub(unresolved_setting, substitutions)
-
-        settings[:masterport] = substitutions['$masterport'] = overrides.fetch(:masterport, '8140')
-
-        settings[:ca_name] =  overrides.fetch(:ca_name, 'Puppet CA: $certname')
-        settings[:root_ca_name] = overrides.fetch(:root_ca_name, "Puppet Root CA: #{SecureRandom.hex(7)}")
+        defaults = {
```

Comment:
Since we never operate on `defaults` as a single collection and the two keys contain collections that are importantly different structurally I could see splitting this into two variables like an Array for `base` and a Hash for `dependent`.

But on the other hand its nice to call them all out as defaults and have the variable names and structure provide clear readability.

I could go either way.

Reply:
Yeah, I tried both in my iterations.  I'll try to see if I can improve it.